### PR TITLE
Added more configuration related tests

### PR DESCRIPTION
**Pull request by ShubhamRwt**
### Type of change

_Select the type of your PR_

- Bugfix
- Enhancement / new feature
- Refactoring
- Documentation

### Description

_Please describe your pull request_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRollingTest.java [hunk 1]
```diff
@@ -344,8 +444,11 @@ public void shouldRestartMultipleBrokersWithTopicWithNoReason() throws Execution
                 3,
                 5);
 
-        // then
```
Comment (tombentley) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRollingTest.java:347]: No need to remove this `// then`, since all the tests are using these comments to delineate the common structure.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRollingTest.java [hunk 2]
```diff
@@ -156,9 +156,6 @@ public void shouldRestartBrokerIfReasonManualRolling() throws ExecutionException
         Mockito.verify(client, times(1)).tryElectAllPreferredLeaders(eq(nodeRef));
     }
 
-
-    // TODO: Make the test when "no partitions on one broker" case
```
Comment (tombentley) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRollingTest.java:160]: Why did you remove this? I still do think we're testing this case, are we?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRollingTest.java [hunk 3]
```diff
@@ -298,6 +295,109 @@ public void shouldReconfigureBrokerWithChangedReconfigurableParameter() throws E
...
+                1);
+
+        // then
+        Mockito.verify(client, never()).reconfigureServer(eq(nodeRef),any(), any());
+        Mockito.verify(client,times(1)).deletePod(eq(nodeRef));
+        Mockito.verify(client, times(1)).tryElectAllPreferredLeaders(eq(nodeRef));
+    }
+
+    @Test
+    public void shouldRestartBrokerWithChangedReconfigurableLoggingParameter() throws ExecutionException, InterruptedException, TimeoutException {
```
Comment (tombentley) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRollingTest.java]: The name of this method is wrong. We're resting reconfig here, not restart.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRollingTest.java (hunk 2 above)
Comment (ShubhamRwt) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/rolling/RackRollingTest.java:160]: I thought it was covered in this test `shouldRestartBrokerIfReasonManualRolling`
